c2rust-analyze/src/rewrite/ty.rs

```diff
@@ -0,0 +1,370 @@
+use crate::context::{AnalysisCtxt, Assignment, LTy};
+use crate::labeled_ty::{LabeledTy, LabeledTyCtxt};
+use crate::rewrite::Rewrite;
+use crate::type_desc::{self, Ownership, Quantity};
+use rustc_ast::ast;
+use rustc_hir as hir;
+use rustc_hir::def::{DefKind, Namespace, Res};
+use rustc_hir::def_id::LocalDefId;
+use rustc_hir::intravisit;
+use rustc_hir::Mutability;
+use rustc_middle::hir::nested_filter;
+use rustc_middle::mir::{self, Body, LocalDecl};
+use rustc_middle::ty::print::{FmtPrinter, Print};
+use rustc_middle::ty::subst::GenericArg;
+use rustc_middle::ty::{self, ReErased, Ty, TyCtxt, TyKind};
+use rustc_span::Span;
+
+/// A label for use with `LabeledTy` to indicate what rewrites to apply at each position in a type.
+#[derive(Clone, Copy, PartialEq, Eq, Debug, Hash)]
+struct RewriteLabel {
+    /// Rewrite a raw pointer, whose ownership and quantity have been inferred as indicated.
+    ty_desc: Option<(Ownership, Quantity)>,
+    /// If set, a child or other descendant of this type requires rewriting.
```

Comment:
what is a child or descendant of a type? including that in the docs somehow (maybe a rustdoc link) would be helpful

Reply:
By "child", I just mean a `Ty` that appears inside of this `Ty`.  For example, the children of `(u32, (u8, u8))` are `u32` and `(u8, u8)`.  "Descendant" is the transitive closure of "child".  I'll update this comment.